pkg/storegateway/series_refs.go

```diff
@@ -876,10 +841,6 @@ func newLoadingSeriesChunkRefsSetIterator(
 	tenantID string,
 	logger log.Logger,
 ) *loadingSeriesChunkRefsSetIterator {
-	if strategy.isNoChunkRefsOnEntireBlock() {
```

Comment:
Can you explain me this change, please? Just a quick pointer.

Reply:
this isn't really necessary; i think i should have included in a different PR that got merged a bit before this one. This just bubbles up the use of the strategy. Instead of using it in `docodeSeries` we call it in `loadingSeriesChunkRefsSetIterator.loadSeries`.

 I thought that passing the strategy is confusing because it's not fully respected. For example if it is `isOverlapMintMaxt() == true`, `decodeSeries` would still return all chunks, not only the chunks that overlap with mint/maxt. Do you think this is better fit for a separate PR?